### PR TITLE
[OpInfo] add reference and error inputs for `multi_margin_loss`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #104578
* #104852
* #104851
* __->__ #104850

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Summary:
- Adds tests for negative `margin` and `reduction`s to sample inputs.
- Adds reference and error inputs.
- Removes redundant tests from other files.
- No tests for `weight` because there are added in subsequent PRs (to make it easier to review).

File: test/test_nn.py [hunk 1]
```diff
@@ -9297,7 +9297,6 @@ def v(fn):
 
             v(lambda: F.nll_loss(input, target, reduction=reduction))
             v(lambda: F.cross_entropy(input, target, reduction=reduction))
-            v(lambda: F.multi_margin_loss(input, target, reduction=reduction))
```
Comment (nkaretnikov) [inline test/test_nn.py:9300]: Invalid reductions are tested in error inputs now.

File: test/test_nn.py [hunk 2]
```diff
@@ -11433,11 +11432,6 @@ def test_batchnorm_update_stats(self, device):
             with torch.backends.cudnn.flags(enabled=False):
                 self._test_batchnorm_update_stats(device)
 
-    def test_multi_margin_loss_errors(self, device):
-        self.assertRaises(RuntimeError,
-                          lambda: nn.functional.multi_margin_loss(torch.randn(5, device=device),
-                                                                  torch.zeros(3, device=device)))
```
Comment (nkaretnikov) [inline test/test_nn.py:11439]: Invalid shapes are tested in error inputs now.

File: test/test_torch.py [hunk 3]
```diff
@@ -732,13 +732,6 @@ def test_scalar_check(self, device):
                     self.assertRaises(RuntimeError,
                                       lambda: torch.nn.functional.multilabel_margin_loss(input, target, reduction='sum'))
 
-        # multi_margin_loss
-        for input in (zero_d, one_d, torch.randn(1, 1, device=device)):
-            for target in (torch.tensor(0, device=device), torch.tensor([0], device=device)):
-                self.assertEqual(target.shape, torch.nn.functional.multi_margin_loss(input, target, reduction='none').shape)
-                self.assertEqual((), torch.nn.functional.multi_margin_loss(input, target, reduction='mean').shape)
-                self.assertEqual((), torch.nn.functional.multi_margin_loss(input, target, reduction='sum').shape)
```
Comment (nkaretnikov) [inline test/test_torch.py:740]: Valid shapes/scalars were already tested in sample inputs.

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -1404,12 +1403,66 @@ def sample_inputs_multi_margin_loss(op_info, device, dtype, requires_grad, **kwa
...
+    yield ErrorInput(SampleInput(make_input(0,), args=(make_input(5,),), kwargs={}),
+                     error_type=RuntimeError,
+                     error_regex=r'Expected non-empty vector or matrix with optional 0-dim batch size, but got: \[0\]')
+    # invalid target
+    yield ErrorInput(SampleInput(make_input(5, 4), args=(make_input(5, 4),), kwargs={}),
+                     error_type=RuntimeError,
+                     error_regex=(
+                         r'inconsistent target size, expected 5 but got \[5, 4\]'
+                         if torch.device(device).type == 'cuda' else
+                         r'inconsistent target size, got: \[5, 4\]'))
```
Comment (nkaretnikov) [inline torch/testing/_internal/common_methods_invocations.py:1457]: This error-message-divergence is fixed in the next PR in the stack.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**